engine/crates/engine-v2/schema/src/walkers/definition.rs

```diff
@@ -106,6 +107,17 @@ impl<'a> DefinitionWalker<'a> {
     pub fn is_object(&self) -> bool {
         matches!(self.item, Definition::Object(_))
     }
+
+    pub fn directives(&self) -> TypeSystemDirectivesWalker<'a> {
+        match self.item {
+            Definition::Scalar(s) => self.walk(s).directives(),
+            Definition::Object(o) => self.walk(o).directives(),
+            Definition::Interface(i) => self.walk(i).directives(),
+            Definition::Union(u) => self.walk(u).directives(),
+            Definition::Enum(e) => self.walk(e).directives(),
+            Definition::InputObject(io) => self.walk(io).directives(),
+        }
```

Comment:
Sometimes I think we are missing a `Definition` table in FederatedGraph.

Reply:
don't think it would be useful, you have a definition abstraction in the composition step. FederatedGraph is a storage format from my perspective.